daml-lf/repl/src/main/scala/com/digitalasset/daml/lf/repl/testing/Main.scala

```diff
@@ -410,31 +410,24 @@ object Repl {
                 submissionTime = Time.Timestamp.now(),
                 transactionSeed = None
               )
-            var count = 0
             val startTime = System.nanoTime()
+            var finished: SValue = null
             var errored = false
-            while (!machine.isFinal && !errored) {
-              machine.run match {
-                case SResultError(err) =>
-                  println(prettyError(err, machine.ptx).render(128))
-                  errored = true
-                case SResultFinalValue(_) => ()
-                case other =>
-                  sys.error("unimplemented callback: " + other.toString)
-              }
-              count += 1
+            machine.run match {
```

Comment:
How about something like `val (finished, errored) = machine.run match { … }`. Mutating state where we don’t have to feels weird.

Reply:
I removed the mutation. The result is now an optional value.

Reply:
Thanks!